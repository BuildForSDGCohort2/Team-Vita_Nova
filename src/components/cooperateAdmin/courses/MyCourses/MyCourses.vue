<template>
  <v-responsive>
    <v-container style="max-width: 1000px" grid-list-md>
      <div :style="{'font-family': 'IBM Plex Sans'}" v-if="renderComponent">
        <v-row class="mt-n5 d-flex">
          <v-col cols="12" md="8">
          </v-col>
          <v-col cols="12" md="4" :style="{'font-size': '18px', 'justify-content': 'flex-end'}">
            <v-btn
                class="btn-course"
                large
                :style="{'font-size': '18px', 'height': '53px', 'text-transform': 'capitalize', 'background': '#FF2E2E', 'color': '#FFF', 'font-weight': 'bold'}"
                @click="forceRerender(false)"
            >
              <v-icon color="#fff">mdi-plus</v-icon>
              Create Course</v-btn>
          </v-col>
        </v-row>
        <v-row>
          <v-col cols="12" md="6">
            <h3 :style="{'text-transform': 'uppercase'}">Drafts</h3>
          </v-col>
        </v-row>
        <v-row>
          <v-col cols="12" md="6" v-if="cards.drafts.length !== 0">
            <div v-for="(card, index) in cards.drafts" :key="index">
              <CourseCard :cards="card"/>
            </div>
          </v-col>
          <v-col cols="12" md="6" v-else-if="cards.publishedCourses.length !== 0">
            <div>
              <h2>You do not have any drafted course</h2>
            </div>
          </v-col>
          <v-col cols="12" md="6" v-else>
            <div>
              <h2>You have not created any course yet</h2>
            </div>
          </v-col>
        </v-row>
        <v-row class="all-courses">
          <v-col cols="12">
            <h3 class="all-courses-header" :style="{'text-transform': 'uppercase'}">Published Courses</h3>
          </v-col>
        </v-row>
        <v-row>
          <v-col cols="12" md="6" v-if="cards.publishedCourses.length !== 0">
            <div v-for="(card, index) in cards.publishedCourses" :key="index">
              <CourseCard :cards="card" :index="index"/>
            </div>
          </v-col>
          <v-col cols="12" md="6" v-else>
            <div>
              <h2>You have not published any course yet</h2>
            </div>
          </v-col>
        </v-row>
      </div>
      <div :style="{'font-family': 'IBM Plex Sans'}" v-else>
        <CreateCourse :method="forceRerender" />
      </div>
    </v-container>
  </v-responsive>
</template>

<script>
// import CourseCard from "@/components/cooperateAdmin/courses/CourseCard";
import CreateCourse from "@/components/cooperateAdmin/courses/MyCourses/CreateCourse";
export default {
  name: "MyCourses",
  data() {
    return {
      renderComponent: true,
      dropdown_font: ['Arial', 'Calibri', 'Courier', 'Verdana'],
      searching: '',
      select: '',
    }
  },
  methods: {
    forceRerender(condition) {
      // Remove my-component from the DOM
      this.renderComponent = condition;
    }
  },
  components: {
    // CourseCard,
    CreateCourse
  },
  computed: {
    cards() {
      return {
        drafts: [
          {
            title: "Introduction to Robotics",
            subtitle: "21 Modules Included",
            company: "Hobo Tech",
            courseImageLink: 'https://images.unsplash.com/photo-1454165804606-c3d57bc86b40?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60',
            rank: 4.8
          }
        ],
        publishedCourses: [
          {
            title: "Nuxt.js for Beginners",
            subtitle: "21 Modules Included",
            courseImageLink: 'https://images.unsplash.com/photo-1481627834876-b7833e8f5570?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60',
            rank: 4.8
          }
        ],
      };
    },
  }
};
</script>

<style scoped>
.all-courses {
  margin-top: 4rem;
}
.all-courses-header {
  color: #2B1C1C;
}
.btn-course {
  /* Auto Layout */
  display: flex;
  flex-direction: row;

  position: static;
  width: 218px;
  height: 63px;
  left: 0;
  top: 0;

  /* Inside Auto Layout */
  flex: none;
  order: 1;
  align-self: center;
  margin: 24px 0;
}
</style>