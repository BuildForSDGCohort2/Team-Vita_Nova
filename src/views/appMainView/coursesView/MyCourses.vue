<template>
  <div>
    <MyCourses />
  </div>
</template>

<script>
import MyCourses from "../../../components/corporateAdmin/courses/myCourses/MyCourses";
export default {
  components: {
    MyCourses
  }
};
</script>