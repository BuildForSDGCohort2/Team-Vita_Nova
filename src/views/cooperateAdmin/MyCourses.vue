<template>
  <div>
    <MyCourses />
  </div>
</template>

<script>
import MyCourses from "../../components/cooperateAdmin/courses/MyCourses/MyCourses";
export default {
  components: {
    MyCourses
  }
};
</script>