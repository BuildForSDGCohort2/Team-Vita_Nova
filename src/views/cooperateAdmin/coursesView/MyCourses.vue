<template>
  <div>
    <MyCourses />
  </div>
</template>

<script>
import MyCourses from "../../../components/cooperateAdmin/courses/myCourses/MyCourses";
export default {
  components: {
    MyCourses
  }
};
</script>