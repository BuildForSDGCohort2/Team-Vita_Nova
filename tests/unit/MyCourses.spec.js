import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";
import Vue from "vue";
import MyCourses from "@/components/cooperateAdmin/courses/MyCourses/MyCourses";

Vue.use(Vuetify)
Vue.config.productionTip = false;
const localVue = createLocalVue();

describe("My Courses Component unit test", () => {
    let vuetify;

    beforeEach(() => {
        vuetify = new Vuetify();
    })

    test("setup correctly", () => {
        expect(true).toBe(true);
    });

    test("render my courses and match snapshot", () => {
        const wrapper = mount(MyCourses, {
            localVue,
            vuetify,
            propsData: {
                title: "MyCourses"
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.text()).toContain("Drafts");
        expect(wrapper.text()).toContain("Published Courses");
    });

    test("render course cards", () => {
        const course = "Introduction to Robotics";
        const wrapper = mount(MyCourses, {
            localVue,
            vuetify,
            propsData: {
                title: "MyCourses",
                cards: {
                    myCourses: [
                        {
                            course: course
                        }
                    ]
                }
            }
        });
        expect(wrapper.html()).toContain(course);
    });

    test("Create course button click", () => {
        const wrapper = mount(MyCourses, {
            localVue,
            vuetify
        });

        const event = jest.fn();
        const button = wrapper.find(".v-btn");
        expect(button.text()).toContain("Create Course")

        button.vm.$on("click", event);
        expect(event).toHaveBeenCalledTimes(0);
        button.trigger("click");
        expect(event).toHaveBeenCalledTimes(1);
    });

});