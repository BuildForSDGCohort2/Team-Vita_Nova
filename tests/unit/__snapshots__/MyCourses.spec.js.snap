// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Courses Component unit test render my courses and match snapshot 1`] = `
<div class="v-responsive" title="MyCourses">
  <div class="v-responsive__content">
    <div class="container grid-list-md" style="max-width: 1000px;">
      <div style="font-family: IBM Plex Sans;">
        <div class="row mt-n5">
          <div class="col-md-9 col-12"></div>
          <div class="col-md-3 col-12" style="font-size: 18px;"><button type="button" class="btn v-btn v-btn--contained theme--light v-size--large" style="font-size: 18px; height: 53px; color: rgb(255, 255, 255); font-weight: bold; background: rgb(255, 46, 46);"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-plus theme--light" style="color: rgb(255, 255, 255); caret-color: #fff;"></i>
            Create Course</span></button></div>
        </div>
        <div class="row">
          <div class="col-md-6 col-12">
            <h3 style="text-transform: uppercase;">Drafts</h3>
          </div>
        </div>
        <div class="row">
          <div class="col-md-6 col-12">
            <div>
              <div>
                <div class="v-card v-sheet theme--light elevation-2 rounded-lg" style="height: 250px; width: 600px; background-size: cover; background-image: url(https://images.unsplash.com/photo-1454165804606-c3d57bc86b40?ixlib=rb-1.2.1&amp;ixid=eyJhcHBfaWQiOjEyMDd9&amp;auto=format&amp;fit=crop&amp;w=500&amp;q=60);">
                  <div class="overlay">
                    <div tabindex="-1" class="v-list-item theme--light">
                      <div class="v-list-item__content">
                        <div tabindex="-1" class="v-list-item theme--light">
                          <div class="v-card__text headline ml-0 mt-5"><a elevation="2" href="Introduction to Robotics" class="" style="color: rgb(255, 255, 255); text-decoration-line: none;">
                              <h1 style="font-family: IBM Plex Sans; font-size: 24px; line-height: 31px;">Introduction to Robotics</h1>
                            </a></div>
                        </div>
                        <div tabindex="-1" class="v-list-item theme--light">
                          <div class="v-card__text prop mt-n10 pr-5" style="font-family: IBM Plex Sans; font-size: 14px;">21 Modules Included</div>
                        </div>
                      </div>
                      <div class="v-list-item__icon">
                        <div class="v-menu"><button type="button" class="mt-7 mr-5 v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default" style="font-family: IBM Plex Sans; font-size: 24px; line-height: 31px;" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-dots-vertical theme--light" style="color: rgb(255, 255, 255); caret-color: #fff;"></i></span></button>
                          <!---->
                        </div>
                      </div>
                    </div>
                    <footer class="v-footer font-weight-medium v-sheet theme--light v-footer--absolute" style="left: 0px; right: 0px; bottom: 0px; color: rgb(255, 46, 46); opacity: 1; background-color: rgba(0, 0, 0, 0);">
                      <div class="text-left col col-12">
                        <div tabindex="-1" absolute="" class="font-weight-medium ml-0 mt-9 v-list-item theme--light"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light" style="color: rgb(255, 255, 255); caret-color: #fff; font-family: IBM Plex Sans; font-size: 14px;"></i>
                          <div class="v-card__text prop" style="font-family: IBM Plex Sans; font-size: 14px;">4.8</div>
                        </div>
                      </div>
                    </footer>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row all-courses">
          <div class="col col-12">
            <h3 class="all-courses-header" style="text-transform: uppercase;">Published Courses</h3>
          </div>
        </div>
        <div class="row">
          <div class="col-md-6 col-12">
            <div>
              <div>
                <div class="v-card v-sheet theme--light elevation-2 rounded-lg" style="height: 250px; width: 600px; background-size: cover; background-image: url(https://images.unsplash.com/photo-1481627834876-b7833e8f5570?ixlib=rb-1.2.1&amp;ixid=eyJhcHBfaWQiOjEyMDd9&amp;auto=format&amp;fit=crop&amp;w=500&amp;q=60);">
                  <div class="overlay">
                    <div tabindex="-1" class="v-list-item theme--light">
                      <div class="v-list-item__content">
                        <div tabindex="-1" class="v-list-item theme--light">
                          <div class="v-card__text headline ml-0 mt-5"><a elevation="2" href="Nuxt.js for Beginners" class="" style="color: rgb(255, 255, 255); text-decoration-line: none;">
                              <h1 style="font-family: IBM Plex Sans; font-size: 24px; line-height: 31px;">Nuxt.js for Beginners</h1>
                            </a></div>
                        </div>
                        <div tabindex="-1" class="v-list-item theme--light">
                          <div class="v-card__text prop mt-n10 pr-5" style="font-family: IBM Plex Sans; font-size: 14px;">21 Modules Included</div>
                        </div>
                      </div>
                      <div class="v-list-item__icon">
                        <div class="v-menu"><button type="button" class="mt-7 mr-5 v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default" style="font-family: IBM Plex Sans; font-size: 24px; line-height: 31px;" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-dots-vertical theme--light" style="color: rgb(255, 255, 255); caret-color: #fff;"></i></span></button>
                          <!---->
                        </div>
                      </div>
                    </div>
                    <footer class="v-footer font-weight-medium v-sheet theme--light v-footer--absolute" style="left: 0px; right: 0px; bottom: 0px; color: rgb(255, 46, 46); opacity: 1; background-color: rgba(0, 0, 0, 0);">
                      <div class="text-left col col-12">
                        <div tabindex="-1" absolute="" class="font-weight-medium ml-0 mt-9 v-list-item theme--light"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light" style="color: rgb(255, 255, 255); caret-color: #fff; font-family: IBM Plex Sans; font-size: 14px;"></i>
                          <div class="v-card__text prop" style="font-family: IBM Plex Sans; font-size: 14px;">4.8</div>
                        </div>
                      </div>
                    </footer>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
